builtin/docker/platform.go

```diff
@@ -101,13 +103,182 @@ func (p *Platform) resourceManager(log hclog.Logger, dcr *component.DeclaredReso
 			resource.WithState(&Resource_Container{}),
 			resource.WithCreate(p.resourceContainerCreate),
 			resource.WithDestroy(p.resourceContainerDestroy),
-
+			resource.WithStatus(p.resourceContainerStatus),
 			resource.WithPlatform(platformName),
 			resource.WithCategoryDisplayHint(sdk.ResourceCategoryDisplayHint_INSTANCE),
 		)),
 	)
 }
 
+func (p *Platform) resourceContainerStatus(
+	ctx context.Context,
+	log hclog.Logger,
+	sg terminal.StepGroup,
+	ui terminal.UI,
+	cli *client.Client,
+	container *Resource_Container,
+	sr *resource.StatusResponse,
+) error {
+	s := sg.Add("Checking status of the docker container resource...")
+	defer s.Abort()
+
+	log.Debug("querying docker for container health")
+
+	// Creating our baseline container resource
+	containerResource := &sdk.StatusReport_Resource{
+		Type:                "container",
+		Platform:            platformName,
+		CategoryDisplayHint: sdk.ResourceCategoryDisplayHint_INSTANCE,
+	}
+
+	// Add the container resource to the the status response. After this function finishes,
+	// the resource manager framework will read the return value out of here.
+	sr.Resources = append(sr.Resources, containerResource)
+
+	// NOTE(briancain): The docker platform currently only deploys a single
+	// container, so for now the status report makes the same assumption.
+	containerInfo, err := cli.ContainerInspect(ctx, container.Id)
+	if err != nil {
+		if client.IsErrNotFound(err) {
+			// We expected this container to be present, but it's not.
+			// It has likely been killed and removed from the platform since it was deployed.
+			containerResource.Name = container.Name
+			containerResource.Id = container.Id
+			containerResource.Health = sdk.StatusReport_MISSING
+		} else {
+			return status.Errorf(codes.FailedPrecondition, "error quering docker for container status: %s", err)
+		}
+	} else {
```

Comment:
Nit: this code has some nested ifs, consider simplifying: 

```go
containerInfo, err := cli.ContainerInspect(ctx, container.Id)
if err != nil {
	if !client.IsErrNotFound(err) {
	  return status.Errorf(codes.FailedPrecondition, "error quering docker for container status: %s", err)
	}
	// We expected this container to be present, but it's not.
	// It has likely been killed and removed from the platform since it was deployed.
	containerResource.Name = container.Name
	containerResource.Id = container.Id
	containerResource.Health = sdk.StatusReport_MISSING
	s.Done()
	return nil
} 

// Add everything that docker knows about the running container to the container resource.
log.Debug("Found docker container", "id", container.Id)
[...]
```